studio/gpu_util.py

```diff
@@ -5,15 +5,48 @@
 from .util import sixdecode
 
 
-def get_available_gpus():
+def memstr2int(string):
+    conversion_factors = [
+        ('Mb', 2**20), ('MiB', 2**20), ('m', 2**20), ('mb', 2**20),
+        ('Gb', 2**30), ('GiB', 2**30), ('g', 2**30), ('gb', 2**30),
+        ('kb', 2**10), ('k', 2**10)
+    ]
+
+    for k, f in conversion_factors:
+        if string.endswith(k):
+            return int(float(string.replace(k, '')) * f)
+
+    return int(string)
+
+
+def get_available_gpus(gpu_mem_needed=None, strict=False):
     gpus = _get_gpu_info()
 
-    def check_gpu(gpu):
+    def check_gpu_nomem_strict(gpu):
         return memstr2int(gpu.find('fb_memory_usage').find('used').text) < \
-            0.1 * memstr2int(gpu.find('fb_memory_usage').find('total').text)
-
-    return [gpu.find('minor_number').text
-            for gpu in gpus if check_gpu(gpu)]
+            memstr2int(gpu.find('fb_memory_usage').find('free').text)
```

Comment:
better logic for handling gpu memory, also new gpuMemStrict flag, which determines how strictly must avaliable memory be a requirement for gpu